lib/beacon/broadcast.go

```diff
@@ -109,7 +110,7 @@ func readBroadcasts(outbox chan<- recv, port int, stop chan struct{}) error {
 	defer close(done)
```

Comment:
`done` should be a child context I think, or potentially same context if there is no other path out the function than via the passed context (I didn't check).

I see there are other places with the same thing, not going to mention them, this might be one of those things you were considering for a followup PR?